### PR TITLE
Document features

### PR DESCRIPTION
**Pull request by brenthuisman**
1. Document neuroscientific features better. Often requested, now urgently needed.
2. Remove stale matplotlib check, addressing https://github.com/arbor-sim/arbor/pull/2157

### CODE REVIEW

File: doc/features/index.rst [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+Features include:
+
+* :ref:`in_gui`: a fully featured GUI for single cell construction.
+* :ref:`Support for a range of fileformats <format-overview>`, including :ref:`formatnmodl` and NeuroML by way of `NMLCC <https://github.com/arbor-sim/nmlcc>`_, a fully featured NeuroML converter for Arbor targets.
+* `Nascent SONATA format support <https://github.com/arbor-sim/arbor-sonata>`_. The Allen mechanism database :ref:`is already included in Arbor <mechanisms_builtins>`, so currently, with some manual work executing Allen models is already possible, see :ref:`tutorialsinglecellallen`.
+* :ref:`Voltage Processes <formatnmodl_voltageproc>` in the form of a mechanism kind. Useful for e.g. implementing voltage clamps.
+* :ref:`Stochastic Differential Equations <tutorial_calcium_stpd_curve>`: both point and density mechanisms may now use white noise as part of the state updates, turning the ODEs effectively into SDEs.
+* :ref:`Mutable connection table <interconnectivity-mut>`: the connection table can be redefined between successive simulation function calls.
```
Comment (thorstenhater) [inline doc/features/index.rst:15]:
```suggestion
* :ref:`Mutable connection table <interconnectivity-mut>`: the connection table can be redefined between successive phases of simulation.
```

File: doc/features/index.rst [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+* :ref:`Edit morphologies <morph-edit>` using Arbor API, useful for e.g. merging segment trees.
+* `BluePyOpt integration <https://github.com/BlueBrain/BluePyOpt/releases/tag/1.14.0>`_. Parameter optimization using BluePyOpt can be done through an Arbor backend and exported to an Arbor cable cell, see :ref:`this tutorial <tutorialsinglecellbluepyopt>`.
+* :ref:`Wide range of cable cell probes<pycablecell-probesample-api>`, :ref:`plus LIF cell probes <pycablecell-probesample-lif>`.
+* :ref:`Inhomogeneous parameters <cablecell-scaled-mechs>`: enables :ref:`painting of scaled density mechanisms <labels-iexpr>`, e.g. with the distance from the root. (nmodl: ``iexpr``.)
+* :ref:`LFPyKit integration <tutorial_lfpykit>`: extract extracellular signals from an Arbor cable cell.
+* :ref:`Faster NMODL programming guide <formatnmodl-faster>`: helps users write NMODL with performance and fewer bugs in mind.
```
Comment (thorstenhater) [inline doc/features/index.rst:21]: That's not really a feature ... maybe we make a list of helpful things a later further on?

File: doc/features/index.rst [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+* :ref:`Axial Diffusion <cablecell-ions-diffusion>`: ions can now propagate along the dendrite by diffusion and be received by other synapses, modifying their weight upon reception.Xd can be read from and written to by NMODL density and point mechanisms. See `this LaTeX file <https://github.com/arbor-sim/arbor/blob/master/doc/dev/axial-diff.tex>`_ for a mathematical description of Arbor's implementation.
+* :ref:`Gap Junction Mechanisms <mechanisms-junction>`.
+* :ref:`Mechanism ABI <extending-catalogues>`, allowing for users to package their mechanism catalogues.
+* :ref:`Built-in profiler <pyprofiler>`, which enables users to quickly understand where their experiment is spending most of its time.
+
+Modelling
+---------
+
+Most of the Arbor community lives in our `Gitter channel <https://gitter.im/arbor-sim/community>`_\, and modellers convene `weekly <https://arbor-sim.org/arbor-weekly-videochat/>`_ to discuss how to use or identify missing features in a vide chat. Please join!
```
Comment (thorstenhater) [inline doc/features/index.rst:30]: Nice!

**Review by thorstenhater — CHANGES_REQUESTED** [review]
Nice, that looks quite polished!

File: doc/features/index.rst (hunk 2 above)
Comment (brenthuisman) [reply doc/features/index.rst:21]: It's indeed not a pure list of neuroscientific features, but I thought it worth mentioning those features that will come in handy when modelling. I'll move this one down at the bottom, where I put others falling in this category.

**Review by thorstenhater — APPROVED** [review]

**Pull request merged**